### PR TITLE
atlas tool: 'update' command

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: tools/atlas/atlas.nim [hunk 1]
```diff
@@ -9,9 +9,11 @@
 ## Simple tool to automate frequent workflows: Can "clone"
 ## a Nimble dependency and its dependencies recursively.
 
-import std/[parseopt, strutils, os, osproc, unicode, tables, sets, json, jsonutils]
+import std/[parseopt, strutils, os, osproc, tables, sets, json, jsonutils]
```
Comment (planetis-m) [inline tools/atlas/atlas.nim:12]:
```suggestion
import std/[parseopt, strutils, os, osproc, unicode, tables, sets, json, jsonutils]
```

File: tools/atlas/atlas.nim [hunk 2]
```diff
@@ -448,6 +452,27 @@ proc installDependencies(c: var AtlasContext; nimbleFile: string) =
...
+      c.withDir file:
+        let pkg = PackageName(file)
+        let (remote, _) = osproc.execCmdEx("git remote -v")
+        if filter.len == 0 or filter in remote:
+          let diff = isCleanGit(c)
+          if diff != "":
+            warn(c, pkg, "has uncommitted changes; skipped")
+          else:
+            let (branch, _) = osproc.execCmdEx("git rev-parse --abbrev-ref HEAD")
+            if branch.strip.len > 0:
```
Comment (planetis-m) [inline tools/atlas/atlas.nim:467]:
```suggestion
            if strutils.strip(branch).len > 0:
```

File: tools/atlas/atlas.nim [hunk 3]
```diff
@@ -448,6 +452,27 @@ proc installDependencies(c: var AtlasContext; nimbleFile: string) =
...
+              let (output, exitCode) = osproc.execCmdEx("git pull origin " & branch.strip)
```
Comment (planetis-m) [inline tools/atlas/atlas.nim:468]:
```suggestion
              let (output, exitCode) = osproc.execCmdEx("git pull origin " & strutils.strip(branch))
```

**Comment by Araq** [comment]
@planetis-m But is the feature itself useful?

**Comment by planetis-m** [comment]
I would say yes as I previously required the 'update' command, but instead, I would re-run atlas clone.

**Pull request merged**